File: javav2/example_code/codecommit/src/main/java/com/example/commit/DeleteBranch.java
// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0

package com.example.commit;

// snippet-start:[codecommit.java2.del_branch.main]
// snippet-start:[codecommit.java2.del_branch.import]
import software.amazon.awssdk.regions.Region;
import software.amazon.awssdk.services.codecommit.CodeCommitClient;
import software.amazon.awssdk.services.codecommit.model.CodeCommitException;
import software.amazon.awssdk.services.codecommit.model.DeleteBranchRequest;
// snippet-end:[codecommit.java2.del_branch.import]

/**
 * To run this Java V2 code example, ensure that you have setup your development
 * environment, including your credentials.
 *
 * For information, see this documentation topic:
 *
 * https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/get-started.html
 */

public class DeleteBranch {
    public static void main(String[] args) {
        final String USAGE = """

                Usage:
                    <repoName> <branchName>\s

                Where:
                    repoName - the name of the repository.
                    branchName - the name of the branch.\s
                """;

        if (args.length != 2) {
            System.out.println(USAGE);
            System.exit(1);
        }

        String repoName = args[0];
        String branchName = args[1];
        Region region = Region.US_EAST_1;
        CodeCommitClient codeCommitClient = CodeCommitClient.builder()
                .region(region)
                .build();

        deleteSpecificBranch(codeCommitClient, repoName, branchName);
        codeCommitClient.close();
    }

    public static void deleteSpecificBranch(CodeCommitClient codeCommitClient, String repoName, String branchName) {
        try {
            DeleteBranchRequest branchRequest = DeleteBranchRequest.builder()
                    .branchName(branchName)
                    .repositoryName(repoName)
                    .build();

            codeCommitClient.deleteBranch(branchRequest);
            System.out.println("The " + branchName + " branch was deleted!");

        } catch (CodeCommitException e) {
            System.err.println(e.getMessage());
            System.exit(1);
        }
    }
}
// snippet-end:[codecommit.java2.del_branch.main]

File: javascriptv3/example_code/codecommit/src/deleteBranch.js
/* Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
SPDX-License-Identifier: Apache-2.0
ABOUT THIS NODE.JS EXAMPLE: This example works with the AWS SDK for JavaScript version 3 (v3),
which is available at https://github.com/aws/aws-sdk-js-v3. For information about how AWS CodeCommit works,
see https://docs.aws.amazon.com/codecommit/latest/userguide/welcome.html.

Purpose:
deleteBranch.js deletes a branch from a repository, unless that branch is the default branch for the repository.

Inputs (replace in code):
- BRANCH_NAME
- REPOSITORY_NAME

Running the code:
node deleteBranch.js
*/

// snippet-start:[codeCommit.JavaScript.deleteBranchV3]
// Get service clients module and commands using ES6 syntax.
import { DeleteBranchCommand } from "@aws-sdk/client-codecommit";
import { codeCommitClient } from "./libs/codeCommitClient.js";

// Set the parameters.

export const params = {
    branchName: 'BRANCH_NAME', /* required */
    repositoryName: 'REPOSITORY_NAME' /* required */
};

// Delete the branch.
export const run = async () => {
    try {
        const data = await codeCommitClient.send(new DeleteBranchCommand(params));
        console.log("Success", data);
        return data; // For unit tests.
    } catch (err) {
        console.log("Error", err);
    }
};
run();
// snippet-end:[codeCommit.JavaScript.deleteBranchV3]

